src/execution_plan/ops/shared/print_functions.c

```diff
@@ -29,7 +30,12 @@ void TraversalToString(const OpBase *op, sds *buf, AlgebraicExpression *ae) {
 											 AlgebraicExpression_Dest(ae));
 	QGEdge *e = (edge) ? QueryGraph_GetEdgeByAlias(op->plan->query_graph, edge) : NULL;
 
-	QGNode_ToString(src, buf);
+	// ignore child label if child is a label-scan.
+	bool ignore_label = (op->type == OPType_CONDITIONAL_TRAVERSE || op->type == OPType_EXPAND_INTO)
+						&& op->children[0]->type == OPType_NODE_BY_LABEL_SCAN;
+	char *label_to_ignore = NULL;
+	if(ignore_label) label_to_ignore = ((NodeByLabelScan *)op->children[0])->n.label;
+	QGNode_ToString(src, buf, label_to_ignore);
```

Comment:
Please extract label information from the Algebraic Expression object,
labels associated with the source node are label matrices at the "source" of the expression, while labels associated with the destination node are found at the "destination"/"end" of the expression.

e.g.
AE = `A * B * ADJ * Z`
`A` and `B` are diagonal label matrices specifying the labels associated with the source node while `Z` is a label attached to the destination node.